### DISCUSSION TITLE
requirements for self-hosted runners

### DISCUSSION THREAD

**Discussion by ipatch** [Homebrew/discussions#4595] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
NA
```

### Output of `brew doctor`

```shell
NA
```

### Description of issue

using a couple of self-hosted runners for obsolete versions of macos ie. catalina and mojave. something i have noticed when using the default github workflow files CI for building bottles for a 3rd party.

- should self-hosted runners already have brew installed? or does it make sense for a self-hosted runner without brew already installed to have the workflow files install and setup brew for the runner?

either way it's not a super big deal just thought i'd ask in case brew is suppose to self setup when using 3rd party taps with self-hosted runners.

my workaround is to ssh into the self-hosted runner and run the install command from brew.sh to get things going.

**Comment by Bo98** [comment, marked as answer]
Ultimately it's up to you and how you prefer to set it up. The installer can work non-interactively, so the answer is go with whatever works best for your setup.

For our own self-hosted runners, what we do is install brew on the runner and make sure it is up-to-date with our `Homebrew/actions/setup-homebrew` action, though our setup also doesn't allow `sudo` that may be necessary to install Homebrew so it's not as if there was a choice for our setup anyway. Also note the `setup-homebrew` action only designed to work when a Homebrew install is already present, at least for now, so if you do install in the workflow you will need to do it before that action if you are using it.

The default 3rd party workflows assume Homebrew is already installed because that's how it is for GitHub-hosted runners.

If your runners are not ephemeral, installing on the machine itself makes sense as you only ever need to do so once. But if it's easier for you to do it in the workflow, then that's fine.

**Reply by ipatch** [reply]
thanks.
